### PR TITLE
Extend unit test coverage in util

### PR DESCRIPTION
**Pull request by koplas**
This is the first batch of new tests that expand the test coverage in the `util` package. Later pull requests will address the unit tests for the `casf` package.

### CODE REVIEW

**Review by JanHoefelmeyer — APPROVED** [review]
Significant improvement in coverage, couldn't find any mistakes:
LGTM & works

File: util/csv_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"bytes"
+	"testing"
+)
+
+func TestCSV(t *testing.T) {
+	buf := new(bytes.Buffer)
+	csvWriter := NewFullyQuotedCSWWriter(buf)
+	for _, x := range [][]string{{"a", "b", "c"}, {"d", "e", "f"}} {
+		err := csvWriter.Write(x)
+		if err != nil {
```
Comment (s-l-teichmann) [inline util/csv_test.go:21]: `if err := csvWriter.Write(x); err != nil {` here. Same on other places, too.

File: util/file_test.go [hunk 2]
```diff
@@ -55,8 +57,54 @@ func TestConformingFileName(t *testing.T) {
 	}
 }
 
-func TestNWriter(t *testing.T) {
+func TestIDMatchesFilename(t *testing.T) {
+	pathEval := NewPathEval()
+
+	doc := make(map[string]interface{})
```
Comment (s-l-teichmann) [inline util/file_test.go:63]: Please use `any` instead of `interface{}`. Same below

File: util/file_test.go [hunk 3]
```diff
@@ -78,3 +126,94 @@ func TestNWriter(t *testing.T) {
...
+		t.Error(err)
+	}
+	if name == name1 {
+		t.Errorf("mkUniq: Expected unique names, got %v and %v", name, name1)
+	}
+}
+
+func TestDeepCopy(t *testing.T) {
+	dir := t.TempDir()
+	os.MkdirAll(filepath.Join(dir, "src/folder0"), 0755)
```
Comment (s-l-teichmann) [inline util/file_test.go:188]: Add error handling.

**Review by s-l-teichmann — CHANGES_REQUESTED** [review]
LGTM (with remarks)

**Review by s-l-teichmann — APPROVED** [review]

**Review by JanHoefelmeyer — APPROVED** [review]
LGTM & works

**Pull request merged**